newapi/ooniapi/database.py

```diff
@@ -58,9 +57,17 @@ def _run_query(query: Query, query_params: dict, query_prio=3):
     settings = {"priority": query_prio, "max_execution_time": 28}
     if isinstance(query, (Select, TextClause)):
         query = str(query.compile(dialect=postgresql.dialect()))
-    q = current_app.click.execute(
-        query, query_params, with_column_types=True, settings=settings
-    )
+    try:
+        q = current_app.click.execute(
+            query, query_params, with_column_types=True, settings=settings
+        )
+    except clickhouse_driver.errors.ServerException as e:
+        title = e.message.splitlines()[0]
+        if title.startswith("DB::Exception"):
+            log.info(e.message)
+            title = title.split(" ", 1)[1]
+        raise Exception(title)
```

Comment:
Since we are already the full database exception stacktrace to our logger, how about we just return a generic error message stating that there was an error with the query?

That's because here we are making an assumption about the way in which stacktraces are formatted by clickhouse and if that doesn't apply anymore in the future (or even in some cases we haven't tested), we might fail open and return more than we should.

Reply:
Ok, updated.